From 466985b2af5507e88ba6318c006fd490734f2fbd Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Mon, 5 Oct 2015 11:54:08 -0400
Subject: [PATCH 0562/1936] Revert "Inhibit messing with the free mode window
 if not in free mode editing."

This reverts commit 9ad6bb7b3c1329dfe4cb5c45ccd7ee9a8c41ac83.
---
 src/UI/EnvelopeFreeEdit.cpp | 6 ------
 src/UI/EnvelopeFreeEdit.h   | 2 --
 2 files changed, 8 deletions(-)

diff --git a/src/UI/EnvelopeFreeEdit.cpp b/src/UI/EnvelopeFreeEdit.cpp
index 7c869692..3c07f5d7 100644
--- a/src/UI/EnvelopeFreeEdit.cpp
+++ b/src/UI/EnvelopeFreeEdit.cpp
@@ -21,7 +21,6 @@ void EnvelopeFreeEdit::init(void)
     oscRegister("Penvdt");
     oscRegister("Penvval");
     oscRegister("Penvsustain");
-    oscRegister("Pfreemode");
 
     //register for non-bulk types
     for(int i=0; i<MAX_ENVELOPE_POINTS; ++i) {
@@ -55,8 +54,6 @@ void EnvelopeFreeEdit::OSC_raw(const char *msg)
         Penvdt[id] = rtosc_argument(msg, 0).i;
     } else if(strstr(msg,"Penvsustain") && !strcmp(args, "i")) {
         Penvsustain = rtosc_argument(msg, 0).i;
-    } else if(strstr(msg,"Pfreemode")) {
-        Pfreemode = !strcmp(args, "T");
     }
     redraw();
     do_callback();
@@ -198,9 +195,6 @@ int EnvelopeFreeEdit::handle(int event)
     const int y_=Fl::event_y()-y();
     static Fl_Widget *old_focus;
 
-    if (!Pfreemode)
-        return 0;
-
     switch(event) {
       case FL_ENTER:
         old_focus=Fl::focus();
diff --git a/src/UI/EnvelopeFreeEdit.h b/src/UI/EnvelopeFreeEdit.h
index ed5ae2c5..9201a901 100644
--- a/src/UI/EnvelopeFreeEdit.h
+++ b/src/UI/EnvelopeFreeEdit.h
@@ -47,6 +47,4 @@ class EnvelopeFreeEdit : public Fl_Box, public Fl_Osc_Widget
         char Penvval[MAX_ENVELOPE_POINTS];
         //The Sustain point
         char Penvsustain;
-        //Are we a free mode editor?
-        char Pfreemode;
 };
-- 
2.47.0

